src/main/java/minicraft/level/tile/farming/PlantTile.java

```diff
@@ -19,8 +20,13 @@ protected PlantTile(String name) {
 
     @Override
     public void steppedOn(Level level, int xt, int yt, Entity entity) {
-        super.steppedOn(level, xt, yt, entity);
+		if (entity instanceof ItemEntity) return;
+		// Very low chance because steppedOn gets called every tick of movement on the tile.
+		if (random.nextInt(200) != 0) return;
         harvest(level, xt, yt, entity);
+		if (random.nextInt(20) == 0)
+			level.setTile(xt, yt, Tiles.get("Dirt"));
+		super.steppedOn(level, xt, yt, entity);
```

Comment:
This is quite redundant as changing into a dirt tile is handled in the super invoke.

Reply:
`FarmTile.steppedOn`'s chances is so low to compensate being called every tick you are moving on a `FarmTile`. But this is called 1 time when the if statement above this doesn't return out of the function, which makes the tile gets harvested.

Reply:
This is called only when the player is moving.

Reply:
Yeah that's why I said every tick that you are moving on the tile...

Reply:
Then, this makes a duplication. At least removing either one of these statements.